modules/tinymce/src/core/main/ts/keyboard/Nbsps.ts

```diff
@@ -203,11 +203,75 @@ const normalizeNbspsInEditor = (editor: Editor): void => {
   }
 };
 
+const startsWithSpaceOrNbsp = (text: string) => /^(&nbsp;|\s)(.*?)/.test(text);
+const endsWithSpaceOrNbsp = (text: string) => /(.*?)(&nbsp;|\s)$/.test(text);
+
+const endsWithDoubleSpaceOrNbsp = (text: string) => /^(.*?)(&nbsp;|\s){2}$/.test(text);
+
+const startsWithDoubleSpaceOrNbsp = (text: string) => /^(&nbsp;|\s){2}(.*?)$/.test(text);
+
+const normalizeNbspWithElements = (content: string, schema: Schema): string => {
+  const wrappedContent = document.createElement('div');
+  wrappedContent.innerHTML = content;
```

Comment:
This doesn't seem like a very performant way to implement filtering. We already have filter processing for paste after content has been parsed into an AST node tree (the logic that `paste_postprocess` uses).

Reply:
removed